pyomo/core/tests/transform/test_add_slacks.py

```diff
@@ -260,28 +260,50 @@ def test_target_objective_create_using(self):
         m = self.makeModel()
         m2 = TransformationFactory('core.add_slack_variables').create_using(
             m, 
-            targets=[ComponentUID(m.rule1), ComponentUID(m.rule3)])
+            targets=[m.rule1, m.rule3])
 
         self.assertFalse(m2.obj.active)
         self.checkTargetsObj(m2)
 
     def test_err_for_bogus_kwds(self):
         m = self.makeModel()
-        OUTPUT = StringIO()
-        with LoggingIntercept(OUTPUT, 'pyomo.core'):
-            TransformationFactory('core.add_slack_variables').apply_to(
-                m,
-                notakwd="I want a feasible model")
-        self.assertIn("Unrecognized keyword arguments in add slack variable "
-                      "transformation:\nnotakwd",
-                      OUTPUT.getvalue())
+        self.assertRaisesRegexp(
+            ValueError,
+            "key 'notakwd' not defined for Config Block*",
+            TransformationFactory('core.add_slack_variables').apply_to,
+            m,
+            notakwd="I want a feasible model"
+            )
+
+    def test_error_for_non_constraint_noniterable_target(self):
+        m = self.makeModel()
+        m.indexedVar = Var([1, 2])
+        self.assertRaisesRegexp(
+            ValueError,
+            "Expected Constraint or list of Constraints.\n\tRecieved "
+            "<class 'pyomo.core.base.var._GeneralVarData'>",
+            TransformationFactory('core.add_slack_variables').apply_to,
+            m,
+            targets=m.indexedVar[1]
+            )
+
+    def test_error_for_non_constraint_target_in_list(self):
+        m = self.makeModel()
+        self.assertRaisesRegexp(
+            ValueError,
+            "Expected Constraint or list of Constraints.\n\tRecieved "
+            "<class 'pyomo.core.base.var.SimpleVar'>",
+            TransformationFactory('core.add_slack_variables').apply_to,
+            m,
+            targets=[m.rule1, m.x]
+            )
```

Comment:
You should add a test that confirms that ComponentUID still works, and that it emits a deprecation warning (probably using LoggingIntercept)